app/models/alchemy/element.rb

```diff
@@ -70,6 +70,8 @@ class Element < BaseRecord
 
     has_many :contents, dependent: :destroy, inverse_of: :element
 
+    before_destroy :destroy_all_nested_elements
```

Comment:
Technically and from a Rails terminology we are deleting and not destroying.

```suggestion
    before_destroy :delete_all_nested_elements
```